### PR TITLE
pyproject.toml dependency and build managment

### PR DESCRIPTION
**Pull request by ekneg54**

### CODE REVIEW

**Review by dtrai2 — CHANGES_REQUESTED** [review]
Looks quite good already. Guess we just have to try it out and see how well it will works. I just found some more mentions of the `requirements.txt` or `requirements_dev.txt` though. Just do a project wide search for those names and you should find them too. Some are for example in the files:
- `doc/source/getting_started.rst`
- `.devcontainer/Dockerfile` (as an optional comment though)
- `.github/workflows/ci.yml`
- `.github/workflows/main.yml`
- `.readthedocs.yaml`
- `README.md`
- ...

**Comment by ekneg54** [comment]
@ppcad @dtrai2 
please have a look again. the pipeline is still failing for black. But I recommend to add the additional 74 format dependent fixes after review :-D

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,122 @@
...
+packages = ["logprep"]
+
+[tool.setuptools.dynamic]
+version = { attr = "logprep.__version__" }
+
+[project]
+name = "logprep"
+description = "Logprep allows to collect, process and forward log messages from various data sources."
+requires-python = ">=3.10"
+# version = "0.1.0"
```
Comment (dtrai2) [inline pyproject.toml:15]:
```suggestion
```

**Review by dtrai2 — CHANGES_REQUESTED** [review]
I just found one small line that you can still also remove. Could you also add an entry in the changelog? The question is are we considereing this as a breaking change? The software itself didn't change, but the install procedure of logprep did, and if someone depends on the `requirements.txt` they have a problem now.

**Review by dtrai2 — APPROVED** [review]

**Pull request merged**